specs/agents/tracing-instrumentation-db.md

```diff
@@ -174,3 +174,22 @@ The following fields are relevant for database and datastore spans. Where possib
 | __**context.destination._**__ |<hr/>|<hr/>|<hr/>|<hr/> |<hr/>|<hr/>|
 |`_.service.name`| `mysql` | `postgresql` | `sqlserver` | `oracle` |  `mariadb` | `db2` |
 |`_.service.resource` | `mysql` | `postgresql` | `sqlserver` | `oracle` |`mariadb` | `db2` |
+
+#### Database instance
+
+For most relational databases, the value of `db.instance` should map to the concept of "current database".
+When no database selected, for example when creating a database, this field should be omitted.
+
+While the semantics may vary across vendors, the goal here is to have a single string that can be used for correlation,
+it is thus important to be able to get the same value across all agents.
+
+There are multiple ways to capture it, agents SHOULD attempt to capture it with the following priorities:
+1. Parsing the database connection string: parsing can be complex, no runtime impact,
+2. Querying connection metadata at runtime: acceptable as fallback, might trigger extra SQL queries, require caching to minimize overhead
+3. Executing a SQL query at runtime: last resort only, require caching to minimize overhead
```

Comment:
I do not like the idea of executing SQL queries against users' databases. I don't think we should use this fallback, but maybe I'm overreacting.

Reply:
No, you are not over-reacting and I think it's fine to remove it from the spec.
I will however keep that for implementation-specific notes as it allows to check if the retrieved value is the one the user expects.